lib/module.js

```diff
@@ -421,38 +421,27 @@ Module._load = function(request, parent, isMain) {
     debug('Module._load REQUEST %s parent: %s', request, parent.id);
   }
 
-  var filename = null;
-
-  if (isMain) {
-    try {
-      filename = Module._resolveFilename(request, parent, isMain);
-    } catch (e) {
-      // try to keep stack
-      e.stack;
-      throw e;
-    }
-    if (experimentalModules) {
-      (async () => {
-        // loader setup
-        if (!ESMLoader) {
-          ESMLoader = new Loader();
-          const userLoader = process.binding('config').userLoader;
-          if (userLoader) {
-            const hooks = await new Loader().import(userLoader);
-            ESMLoader.hook(hooks);
-          }
+  if (isMain && experimentalModules) {
+    (async () => {
+      // loader setup
+      if (!ESMLoader) {
+        ESMLoader = new Loader();
+        const userLoader = process.binding('config').userLoader;
+        if (userLoader) {
+          const hooks = await new Loader().import(userLoader);
+          ESMLoader.hook(hooks);
         }
-        await ESMLoader.import(getURLFromFilePath(filename).href);
-      })()
-      .catch((e) => {
-        console.error(e);
-        process.exit(1);
-      });
-      return;
-    }
-  } else {
-    filename = Module._resolveFilename(request, parent, isMain);
+      }
+      await ESMLoader.import(getURLFromFilePath(request).href);
+    })()
+    .catch((e) => {
+      console.error(e);
+      process.exit(1);
+    });
+    return;
   }
+  
+  var filename = Module._resolveFilename(request, parent, isMain);
```

Comment:
Why do you use `var`?

Reply:
To match the code beneath it goes with - this is just moving the existing line from https://github.com/nodejs/node/pull/16147/files#diff-d1234a869b3d648ebfcdce5a76747d71L424. Although it could be refactored in the process certainly.